core-backend/sandbox/src/funcs.rs

```diff
@@ -338,7 +339,10 @@ where
         syscall_trace!("alloc", pages);
 
         ctx.run_any(RuntimeCosts::Alloc, |ctx| {
-            // TODO: return u32::MAX in case this is error #2353
+            if pages > MAX_WASM_PAGE_COUNT as u32 {
```

Comment:
No, we need to return error value in case line
```rust
let pages = WasmPage::new(pages).map_err(|_| TrapExplanation::Unknown)?;
```
returns an error.
But better is to handle it with `res` below:
```rust
            let pages = WasmPage::new(pages).map_err(|_| TrapExplanation::Unknown)?;

            let res = ctx.ext.alloc(pages, &mut ctx.memory);
            let res = ctx.process_alloc_func_result(res)?;
```
change to
```rust
            let res = (|| {
                   let pages = WasmPage::new(pages).map_err(|_| ...)?;
                   ctx.ext.alloc(pages, &mut ctx.memory)
            })();
            let res = ctx.process_alloc_func_result(res)?;
```